### DISCUSSION TITLE
Request: Add rate() functions to the alerting toolkit so alerts can be fired based on the slope of a line

### DISCUSSION THREAD

**Discussion by pritster5** [grafana/grafana#64175] [Alerting]
Shouldn't be too difficult to add. It would be extremely useful to fire an alert based on the rate of change of data overtime, as threshold based alerting doesnt apply in many cases. 

E.g.:

![image](https://user-images.githubusercontent.com/7132319/222810987-928a58bd-0b79-4757-9ed9-ceef8bd13105.png)

In the image above, I have an SQS queue displaying it's "Approximate Number of Messages Visible" metric. I would like to be alerted if the slope of the line is not negative over an extended period of time (say for 1hr).

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82844. Feel free to continue the discussion around this there. Thank you!
